src/promptflow/promptflow/exceptions.py

```diff
@@ -75,8 +75,16 @@ def message_parameters(self):
         if not self._kwargs:
             return {}
 
-        arguments = self.get_arguments_from_message_format(self.message_format)
-        return {k: v for k, v in self._kwargs.items() if k in arguments}
+        required_arguments = self.get_arguments_from_message_format(self.message_format)
+        parameters = {}
+        for argument in required_arguments:
+            if argument not in self._kwargs:
+                # Set one default value for the missing argument to avoid KeyError.
+                # For long term solution, use CI to guarantee the message_format and message_parameters are in sync.
+                parameters[argument] = "N/A"
+            else:
+                parameters[argument] = self._kwargs[argument]
+        return parameters
```

Comment:
Instead of appending missing arguments into `message_parameters`, it might be better to move this part into the format logic of `message`.